### PR TITLE
[1.16.5] Converted Skin Customization Screen ASM to Mixin

### PR DESCRIPTION
**Pull request by JacksonPlayz**
This PR changes the Slabfish Hat customization button ASM to use Mixins instead.

### CODE REVIEW

File: src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public abstract class CustomizeSkinScreenMixin extends SettingsScreen {
+
+	public CustomizeSkinScreenMixin(Screen previousScreen, GameSettings gameSettingsObj, ITextComponent textComponent) {
+		super(previousScreen, gameSettingsObj, textComponent);
+	}
+
+	@ModifyVariable(method = "init", ordinal = 0, at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/CustomizeSkinScreen;addButton(Lnet/minecraft/client/gui/widget/Widget;)Lnet/minecraft/client/gui/widget/Widget;", ordinal = 1, shift = At.Shift.AFTER))
+	public int init(int i) {
+		++i;
+		this.addButton(new Button(this.width / 2 - 155 + i % 2 * 160, this.height / 6 + 24 * (i >> 1), 150, 20, new TranslationTextComponent(SlabfishHatScreen.SLABFISH_SCREEN_KEY), (button) -> this.getMinecraft().displayGuiScreen(new SlabfishHatScreen(this)), (button, stack, mouseX, mouseY) -> this.renderTooltip(stack, this.getMinecraft().fontRenderer.trimStringToWidth(new TranslationTextComponent(SlabfishHatScreen.SLABFISH_SCREEN_KEY + ".tooltip", new StringTextComponent("patreon.com/teamabnormals").modifyStyle(style -> style.setColor(Color.fromHex("#FF424D")))), 200), mouseX, mouseY)));
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java:27]: `this.getMinecraft()` could be moved to a local variable.

File: src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import net.minecraft.util.text.Color;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+
+@Mixin(CustomizeSkinScreen.class)
+public abstract class CustomizeSkinScreenMixin extends SettingsScreen {
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java:18]: Does this need to be `abstract`?
`final` would work better, in my opinion.

File: src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+	public CustomizeSkinScreenMixin(Screen previousScreen, GameSettings gameSettingsObj, ITextComponent textComponent) {
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java:20]:
```suggestion
	private CustomizeSkinScreenMixin(Screen previousScreen, GameSettings gameSettingsObj, ITextComponent textComponent) {
```

**Review by SmellyModder — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java (hunk 2 above)
Comment (JacksonPlayz) [reply src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/client/CustomizeSkinScreenMixin.java:18]: I'm just used to making all my mixins abstract, can do 👍

**Review by SmellyModder — APPROVED** [review]
ASM is gone :crab:

**Pull request merged**